src/azul/service/storage_service.py

```diff
@@ -206,9 +210,20 @@ def __abort(self):
         self.thread_pool.shutdown(wait=False)
         logger.warning('Upload %s: Aborted', self.upload_id)
 
+    def _submit(self, fn, *args, **kwargs):
```

Comment:
Please give attribution to where we took this idea (and almost verbatim implementation) from. Credit, where credit is due.